omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_postgres_validator.rb

```diff
@@ -190,7 +193,7 @@ def backend_verify_postgres_version(connection)
     # Note that we're looking for the same major, and using our minor as the minimum version
     # This provides compatibility with external databases that use < 9.6 before we make use
     # of any features available in > 9.2.
-    unless (major == REQUIRED_MAJOR) && (minor >= REQUIRED_MINOR)
+    unless SUPPORTED_MAJOR.include?(major) || ((major == REQUIRED_MAJOR) && (minor >= REQUIRED_MINOR))
       fail_with err_CSPG014_bad_postgres_version(v)
```

Comment:
this will need to be a warning not a failure.
anything < 9.2 - fail
9.2 < version < 13.3 -  warn unsupported versions

Reply:
What's the warn command?

Reply:
```
mac:chef-server lbaker$ find . -type f -name "*rb" -exec grep warn_ {} \; -print
  context '#warn_about_removed_attribute' do
        postgres_validator.warn_about_removed_attribute('foo')
        postgres_validator.warn_about_removed_attribute('foo')
./omnibus/files/private-chef-cookbooks/private-chef/spec/libraries/preflight_postgres_validator_spec.rb
        warn_if_cred_mismatch(secret_spec)
    def warn_if_cred_mismatch(opts)
./omnibus/files/private-chef-cookbooks/private-chef/libraries/private_chef.rb
    warn_about_removed_attribute('checkpoint_segments')
  def warn_about_removed_attribute(attr)
./omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_postgres_validator.rb
      def warn_on_incorrect_clients_group(dir, op)
./dev/dotfiles/knife-ec-backup/lib/chef/knife/ec_base.rb
        warn_on_incorrect_clients_group(dest_dir, "backup")
./dev/dotfiles/knife-ec-backup/lib/chef/knife/ec_backup.rb
        warn_on_incorrect_clients_group(dest_dir, "restore")
./dev/dotfiles/knife-ec-backup/lib/chef/knife/ec_restore.rb
```

Reply:
what's the warn command?  see ^^^

Reply:
something along the lines of: https://github.com/chef/chef-server/blob/0566fa10cd47776e36b4dbde75c2bcf28a72ca03/omnibus/files/private-chef-cookbooks/private-chef/libraries/preflight_postgres_validator.rb#L104

Reply:
Added new preflight checks:
PG 13.3 passes.
9.2 < version < 13.3 warns unsupported version.
Tests:
1) New preflight checks pass verify tests.
2) A new test branch with the NEGATE of the new preflight checks fails verify tests.
https://buildkite.com/chef/chef-chef-server-master-verify/builds/4685#a3d3c8bc-8da1-4a82-b8f9-7f26d81852bd